sympy/functions/elementary/integers.py

```diff
@@ -343,3 +349,85 @@ def _eval_Eq(self, other):
             if (self.rewrite(floor) == other) or \
                     (self.rewrite(ceiling) == other):
                 return S.true
+            # Check if other < 0
+            if other.is_extended_real and other.is_extended_negative:
+                return S.false
+            # Check if other >= 1
+            res = self._value_one_or_more(other)
+            if res is not None:
+                return S.false
+
+    def _eval_is_finite(self):
+        return True
+
+    def _eval_is_real(self):
+        return self.args[0].is_extended_real
+
+    def _eval_is_imaginary(self):
+        return self.args[0].is_imaginary
+
+    def _eval_is_integer(self):
+        return self.args[0].is_integer
+
+    def _eval_is_zero(self):
+        return fuzzy_or([self.args[0].is_zero, self.args[0].is_integer])
+
+    def _eval_is_negative(self):
+        return False
+
+    def __ge__(self, other):
+        if self.is_extended_real:
+            other = _sympify(other)
+            # Check if other <= 0
+            if other.is_extended_real and other.is_extended_nonpositive:
```

Comment:
Maybe it is enough to just check `other.is_extended_nonpositive` here?

Reply:
I think so. You could confirm by see if `Dummy(extended_nonpositive=True).is_extended_real` is true.

Reply:
This is correct. If extended_nonpositive is true then extended_real will always be True automatically. This should probably be changed.

In principle there can be a speed difference though between checking the assumptions in a different order since `is_x` will call `_eval_is_x` first before trying other `_eval_is_y` methods to make the inference. I doubt that's worth worrying about though.